Replace OreSatId with Consts from oresat-configs

ThirteenFish opened the pull request:
A while back I deprecated OreSatId in favor of Consts, which provides a more feature rich version of OreSatId. This uses those features here in action, `from_string()` replaces argument handling for naming missions, `default()` ensures tests are run with the current mission, and `__str__()` provides a standard human readable name for the rest api.

This also fixes bitrot in the CI.

ThirteenFish committed:
Replace OreSatId with Consts from oresat-configs

A while back I deprecated OreSatId in favor of Consts, which provides a
more feature rich version of OreSatId. This uses those features here in
action, `from_string()` replaces argument handling for naming missions,
`default()` ensures tests are run with the current mission, and
`__str__()` provides a standard human readable name for the rest api.

Getting the Mission via id from the od was harder than it needs to be,
in the next oresat-configs update will introduce a `Consts.from_id()`
similar to `Consts.from_string()` to make this easier.

ThirteenFish committed:
Delete TimerLoop

For whatever reason pylint is failing CI on this class now when it
worked 6 months ago. Searching through the code though this doesn't seem
to be used at all anymore? As far as I can tell it's been replaced with
other more specific looping things.

ThirteenFish committed:
Clean up CI, Fix breakages

This cleans out the tests action, since oresat-configs is on pypi it
doesn't need to be hand built anymore, and also updates the actions to
stop warnings on the runs. It also fixes:
- A new(?) lint introduced by pylint that complains of too many args to
  sdo_write_bitfield
- mypy type erroring on the updater test. In 829de290 updates_cached()
  was changed to return a list instead of an int, but the corresponding
  test was never updated. The test was never run because it's only set
  to run if being executed as root, which we never do.

ThirteenFish committed:
Haha it turns out that function exists

In d5c41a6 I complained that past me hadn't thought to write
`Consts.from_id()` but it turns out I did write it and I deleted it on
accident two hours ago on my local copy of oresat-configs while doing
other dev work. Oops.

Anyway this is much nicer.

ryanpdx commented on .github/workflows/tests.yaml:
```diff
@@ -10,36 +10,17 @@ on:
-    - name: Clone oresat-c3 repository
-      uses: actions/checkout@v2
-
-    - name: Clone oresat-configs repository
```
It is better to build the latest version of `oresat-configs`, as the last release may not have changes needed by this project. Main case is where new a object are added to the OD that the code now uses.

ThirteenFish replied:
Would it make sense to do small `oresat-configs` point releases at a quicker cadence in that case? My concern is that while you and I both can easily and know to install a dev version of `oresat-configs`, someone starting on the project is going to clone the repo and `pip install` the dependencies and then get stuck because it doesn't work.

ryanpdx replied:
I can see where that could be confusing, but we can point to the script in the `oresat-configs` repo for the fix.

I don't think having a quicker cadence wouldn't help, as adding new code that depends on a new object would then require a new release just for the workflow to pass. Making a new release each time an object is added little much. Waiting for the next release isn't great either.

ThirteenFish committed:
Partial revert: "Clean up CI, Fix breakages"

This partially reverts 9a880678a4be95187a020fd072ff18ac8a5dab45.

From Ryan:
> It is better to build the latest version of oresat-configs, as the
last release may not have changes needed by this project. Main case is
where new a object are added to the OD that the code now uses.

ThirteenFish committed:
Use Mission instead of Consts

In the latest oresat-configs update Consts was renamed to Mission.
